Jenkinsfile

```diff
@@ -1,27 +1,78 @@
+@Library('pipeline-library') _
 pipeline {
+    agent {
+        node{
+            label "dev_mesos_large_v2"
+        }
+    }
+
+    environment {
+        CREDENTIALS_ID  = "GENESYSCLOUD_OAUTHCLIENT_ID_AND_SECRET"
+        GOPATH = "$HOME/go"
+		TF_ACC = "1"
+        TF_LOG = "DEBUG"
+        TF_LOG_PATH = "../test.log"
+		GENESYSCLOUD_REGION = "us-east-1"
+        GENESYSCLOUD_SDK_DEBUG =  "true"
+        GENESYSCLOUD_TOKEN_POOL_SIZE =  20
+    }
+    tools {
+        go 'Go 1.20'
+        terraform 'Terraform 1.0.10'
+    }
+
+    stages {
+      
+        
+        stage('Install Dependencies & Build') {
+            steps {
+                echo 'Installing dependencies'
+                sh 'go version'
+                sh 'go mod download'
+                sh 'go build -v .'
+            }
+	    }
+
+        stage('Terraform Check') {
+            steps {
+                echo 'Check Terraform Installation'
+                sh 'terraform -version'
+
+            }
+        }
+        
+        stage('Tests') {
+            steps {
+            
+                catchError(buildResult: 'SUCCESS', stageResult:'FAILURE'){
+                    echo 'Attempting to Run Tests'
+                    withCredentials([usernamePassword(credentialsId: CREDENTIALS_ID, usernameVariable: 'GENESYSCLOUD_OAUTHCLIENT_ID',passwordVariable:'GENESYSCLOUD_OAUTHCLIENT_SECRET')])
+                        {
+                            echo 'Loading Genesys OAuth Credentials'
+                            sh 'go test -timeout 80m -v -cover ./genesyscloud/... -parallel 8 -coverprofile=coverage.out'
+                            
+                        }
 
-  agent {
-        node {
-            label 'dev_v2'
+                }
+
+            }
         }
-  }
-
-  stages {
-    stage ('Release') {
-      environment {
-        GITHUB_TOKEN = credentials('MYPURECLOUD_GITHUB_TOKEN')
-        GPG_FINGERPRINT="25D753B7C560659B057B714C970A8360B4BF5075"
-      }
-
-      steps {
-         withCredentials([file(credentialsId: 'TERRAFORM_GPG', variable: 'terraform_gpg_private_key')]) {
-                    sh "cp \$terraform_gpg_private_key /tmp/terraform_gpg_secret.asc & chmod 755 /tmp/terraform_gpg_secret.asc"
-                    sh "./addCredToConfig.sh"
-                    sh "rm -f secret.asc"
+
+        stage('Generate Readable Coverage Report') {
+            steps {
+                sh 'go tool cover -html coverage.out -o coverage.html'
+
+            }
         }
 
-        sh './getgoreleaser.sh release --clean --timeout 45m --parallelism 3'
-      }
-    }
-  }
+
+        stage('Upload artifacts') {
+            steps {
+                archiveArtifacts artifacts: 'coverage.html', allowEmptyArchive: true
+
+            }
+        }
+
```

Comment:
Is the release stage supposed to be gone? I think it is missing from the file now